### DISCUSSION TITLE
Foundry compiler error when using cheatcodes

### DISCUSSION THREAD

**Discussion by stmario** [foundry-rs/foundry#1569] [Help] [answered]
I have the following simple setup and somehow some cheatcodes just won't work, even though they are present and declared `public` in `Test.sol`.
```
pragma solidity ^0.8.13;

import "../lib/forge-std/src/Test.sol";

contract MyTest is Test {

    function setUp() public {
        skip(5); // somehow works
        // hoax(0xdeadbeef); Doesn't work: TypeError: No matching declaration found after argument-dependent lookup.
        //vm.prank(0xdeadbeef); Doesn't work: TypeError: Member "prank" not found or not visible after argument-dependent lookup in contract Vm.
    }
}
```
Thanks for your help. `forge test` works apart from this issue.

**Comment by mds1** [comment, marked as answer]
You just need to cast to address first, i.e. `hoax(address(0xdeadbeef))` and `vm.prank(address(0xdeadbeef))`. I think the default type it gives without casting is an integer

**Reply by stmario** [reply]
What? I swear I remember trying that. Yes, it works thank you!
